atomic_reactor/plugins/post_pulp_sync.py

```diff
@@ -0,0 +1,169 @@
+                 username=None, password=None,
+                 dockpulp_loglevel=None):
+        """
+        constructor
+
+        :param tasker: DockerTasker instance
+        :param workflow: DockerBuildWorkflow instance
+        :param pulp_registry_name: str, name of pulp registry to use,
+               specified in /etc/dockpulp.conf
+        :param docker_registry: str, URL of docker registry to sync from
+        :param pulp_secret_path: path to pulp.cer and pulp.key
+        :param username: pulp username, used in preference to
+               certificate and key
+        :param password: pulp password, used in preference to
+               certificate and key
+        """
+        # call parent constructor
+        super(PulpSyncPlugin, self).__init__(tasker, workflow)
+        self.pulp_registry_name = pulp_registry_name
+        self.docker_registry = docker_registry
+        self.pulp_secret_path = pulp_secret_path
+        self.username = username
+        self.password = password
+
+        if dockpulp_loglevel is not None:
+            logger = dockpulp.setup_logger(dockpulp.log)
+            try:
+                logger.setLevel(dockpulp_loglevel)
+            except (ValueError, TypeError) as ex:
+                self.log.error("Can't set provided log level %r: %r",
+                               dockpulp_loglevel, ex)
+
+    def set_auth(self, pulp):
+        if self.username and self.password:
+            # Use username and password if provided
+            pulp.login(self.username, self.password)
+        elif self.pulp_secret_path:
+            # Work out the pathnames for the certificate/key pair
+            cer = os.path.join(self.pulp_secret_path, self.CER)
+            key = os.path.join(self.pulp_secret_path, self.KEY)
+
+            if not os.path.exists(cer):
+                raise RuntimeError("Certificate does not exist")
+            if not os.path.exists(key):
+                raise RuntimeError("Key does not exist")
+
+            # Tell dockpulp
+            pulp.set_certs(cer, key)
+
+    def run(self):
+        pulp = dockpulp.Pulp(env=self.pulp_registry_name)
+        self.set_auth(pulp)
+
+        # We only want the hostname[:port]
+        pulp_registry = re.sub(r'^https?://([^/]*)/?.*',
+                               lambda m: m.groups()[0],
+                               pulp.registry)
+
+        # Store the registry URI in the push configuration
+        self.workflow.push_conf.add_pulp_registry(self.pulp_registry_name,
+                                                  pulp_registry)
+
+        images = []
+        self.log.info("syncing from docker V2 registry %s",
+                      self.docker_registry)
+        with dockpulp_config(docker_registry=self.docker_registry) as config:
+            for image in self.workflow.tag_conf.primary_images():
+                self.log.info("syncing %s", image.pulp_repo)
+                pulp.syncRepo(config.env, image.pulp_repo,
```

Comment:
Oh, maybe you mean: 'does sync only copy over newer tags or also delete tags not present in the docker registry?'